src/api/useAuth.tsx

```diff
@@ -20,22 +22,41 @@ export const SignInButton = () => (
 /**
  * A button that signs the user out of the app using Firebase Auth.
  */
-export const SignOutButton = () => (
-	<button type="button" onClick={() => auth.signOut()}>
-		Sign Out
-	</button>
-);
+export const SignOutButton = () => {
+	const navigate = useNavigate();
+
+	return (
+		<button
+			type="button"
+			onClick={() => {
+				auth
+					.signOut()
+					.then(() => {
```

Comment:
You shouldn't have to do this because `ProtectedRoute` is doing its job very well 😄 it can be a double edged sword doing this imo. On one hand, if we weren't able to `subscribe` to events like we can with Firebase, you'd _have_ to do this; in fact, I generally prefer seeing very "direct" actions like this because it makes it very easy to reason with the application. That being said, because the app doesn't _need_ this code to achieve the functionality, it can actually work against our goal of making this action "direct"; having this here suggests that this could be changed to change the sign out direct functionality, but that's not the case: it's adding complexity. You should be safe to remove this 🙏